internal/pkg/bulk/engine.go

```diff
@@ -198,29 +199,18 @@ func (b *Bulker) CreateAndGetBulker(ctx context.Context, zlog zerolog.Logger, ou
 var newESClient = es.NewClient
 
 func (b *Bulker) createRemoteEsClient(ctx context.Context, outputName string, outputMap map[string]map[string]interface{}) (*elasticsearch.Client, error) {
-	hostsObj := outputMap[outputName]["hosts"]
-	hosts, ok := hostsObj.([]interface{})
-	if !ok {
-		return nil, fmt.Errorf("failed to get hosts from output: %v", hostsObj)
-	}
-	hostsStrings := make([]string, len(hosts))
-	for i, host := range hosts {
-		hostsStrings[i], ok = host.(string)
-		if !ok {
-			return nil, fmt.Errorf("failed to get hosts from output: %v", host)
-		}
+	var esOutput config.Elasticsearch
+	esConfig, err := ucfg.NewFrom(outputMap[outputName], config.DefaultOptions...)
+	if err != nil {
+		return nil, err
 	}
-	serviceToken, ok := outputMap[outputName]["service_token"].(string)
-	if !ok {
-		return nil, fmt.Errorf("failed to get service token from output: %v", outputName)
+	err = esConfig.Unpack(&esOutput)
```

Comment:
should there be validation that the required fields are in the config, or is that done by the schema?

Reply:
Added back the validation, the scema will only validate that data is correctly formed